kernel/Cargo.toml

```diff
@@ -39,27 +39,29 @@ uuid = "1.10.0"
 z85 = "3.0.5"
 
 # bring in our derive macros
-delta_kernel_derive = { path = "../derive-macros", version = "0.6.1" }
+delta_kernel_derive = { path = "../derive-macros", version = "0.7.0" }
```

Comment:
```suggestion
delta_kernel_derive = { path = "../derive-macros", version = "0.6.1" }
```

Reply:
````
error: failed to select a version for the requirement `delta_kernel_derive = "^0.6.1"`
candidate versions found which didn't match: 0.7.0
location searched: /home/tyler/source/github/delta-io/delta-kernel-rs/derive-macros
required by package `delta_kernel v0.7.0 (/home/tyler/source/github/delta-io/delta-kernel-rs/kernel)`
```

I promise I didn't make this change willy nilly :smile:  The [multiple locations](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations) support in Cargo requires that the version matches the local dependency in order for it to be used, and the local version is 0.7.0 because the version is defined in the workspace and inherited by `derive-macros/Cargo.toml`

Reply:
oh but i meant to revert everything back to `0.6.1` - does this work if you revert back the workspace to `0.6.1`?

Reply:
@zachschuermann this is a breaking change, are you asking because you want the version change to happen separately? This could not be safely released as a 0.6.1

Reply:
Yeah. Our MO so far has been, leave the old version number until we decide to release a new version. Then one PR that updates the version and adds a changelog, tag and release that.

That's how I've always done releases, but if there is a "better" way to do it, happy to discuss changing the process.